mithril.js

```diff
@@ -35,6 +35,45 @@ var m = (function app(window, undefined) {
 		return VERSION;
 	};
 
+    function slice(x){
+        return Array.prototype.slice.call(x)
+    }
+
+    function getAttributes(node) {
+        return slice(node.attributes).reduce(function(a, v){
+            if(v.name === 'class'){
+                a.className = a.class
+            } else {
+                a[v.name] = v.value
+            }
+            return a;
+        }, {})
+    }
+
+    function rebuild(node){
+        if(node instanceof NodeList){
+            return Array.prototype.slice.call(node).map(rebuild)
+        } else if(node instanceof Text){
+            return {
+                children: [String(node.textContent)],
+                nodes: [node],
+                attrs: {},
+                intact: true
+            }
+        } else {
+            nodeCache.push(node)
+            var r = {
+                tag: node.tagName.toLowerCase(),
+                attrs: getAttributes(node),
+                children: rebuild(node.childNodes),
+                nodes: [node],
+                intact: true
+            }
+            r.children.nodes = Array.prototype.slice.call(node.childNodes)
+            return r
+        }
+    }
```

Comment:
Some of the unit tests are failing because `configContext` is not being set on any of the cached nodes. You'll need a way to get the `config` functions and add them.